### DISCUSSION TITLE
Automated Hosts Import

### DISCUSSION THREAD

**Discussion by fzebib** [zabbix-community/helm-zabbix#50] [Q&A] [answered]
Hi all, I was wondering if there was a good way to import a list of hosts during bootstrap/adding the values via the json/xml/yaml? Would love to add this to part of the configurations/automation, rather then manually importing a list of hosts. Thanks!

**Comment by aeciopires** [comment, marked as answer]
Hello @fzebib!

Zabbix 6.4 has this feature:

- https://www.zabbix.com/documentation/current/en/manual/xml_export_import

By default, Zabbix can import host and other configurations using XML file: https://www.youtube.com/watch?v=sRlY9Rte3FI

But you can develop your scripts of the import/export using the Zabbix API. You need learning a programming language and Zabbix API to create your scripts of the import/export custom.

- https://www.zabbix.com/documentation/current/en/manual/api 
- https://blog.zabbix.com/zabbix-api-the-easy-way/11748/
- https://github.com/zabbix/python-zabbix-utils/blob/main/README.md
- https://www.zabbix.com/forum/zabbix-help/412432-how-to-import-multiple-hosts-using-script-or-a-csv-file
- https://github.com/intellitrend/zabbix-csv-host-import-module
